include/bluetooth/audio/has.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_HAS_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_HAS_H_
+
+/**
+ * @brief Hearing Access Service (HAS)
+ *
+ * @defgroup bt_has Hearing Access Service (HAS)
+ *
+ * @ingroup bluetooth
+ * @{
+ *
+ * The Hearing Access Service is used to identify a hearing aid and optionally
+ * to control hearing aid presets.
+ *
+ * [Experimental] Users should note that the APIs can change as a part of
+ * ongoing development.
+ */
+
+#include <bluetooth/conn.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Opaque Hearing Access Service object. */
+struct bt_has;
+
+/** Hearing Aid device type */
+enum bt_has_hearing_aid_type {
+	BT_HAS_HEARING_AID_TYPE_BINAURAL,
+	BT_HAS_HEARING_AID_TYPE_MONAURAL,
+	BT_HAS_HEARING_AID_TYPE_BANDED,
+};
+
+/** Hearing Aid device capablilities */
+enum bt_has_capabilities {
+	BT_HAS_PRESET_SUPPORT = BIT(0),
+};
+
+/** @brief Hearing Access Service Client callback structure. */
+struct bt_has_client_cb {
+	/**
+	 * @brief Callback function for bt_has_discover.
+	 *
+	 * This callback is called when discovery procedure is complete.
+	 *
+	 * @param conn Bluetooth connection object.
+	 * @param err 0 on success, ATT error or negative errno otherwise.
+	 * @param has Pointer to the Hearing Access Service object or NULL on errors.
+	 * @param type Hearing Aid type.
+	 * @param caps Hearing Aid capabilities.
+	 */
+	void (*discover)(struct bt_conn *conn, int err, struct bt_has *has,
+			 enum bt_has_hearing_aid_type type, enum bt_has_capabilities caps);
+
+	/**
+	 * @brief Callback function for Hearing Access Service active preset.
+	 *
+	 * Called when the value is changed by the remote server.
+	 *
+	 * @param has Pointer to the Hearing Access Service object.
+	 * @param index Active preset index.
+	 */
+	void (*active_index)(struct bt_has *has, uint8_t index);
+};
+
+/** @brief Registers the callbacks used by the Hearing Access Service client.
+ *
+ *  @param cb The callback structure.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_has_client_cb_register(const struct bt_has_client_cb *cb);
```

Comment:
I see that `bt_has_client_cb_register` allows `cb` to NULL. Is this by design to allow an unregister? If so, that should be documented. If not intentional, please add a check for `cb == NULL`.

Reply:
OK. I see none of the rest of the profiles have it documented not have the check.